sql/src/main/java/io/crate/jobs/AbstractExecutionSubContext.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.jobs;
+
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.logging.Loggers;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.CancellationException;
+
+public abstract class AbstractExecutionSubContext implements ExecutionSubContext, ExecutionState {
+
+    protected static final ESLogger LOGGER = Loggers.getLogger(AbstractExecutionSubContext.class);
+
+    protected final SubExecutionContextFuture future = new SubExecutionContextFuture();
+    private final int id;
+
+    protected JobExecutionContext context;
+
+    // TODO: remove this flag when ExecutionState is obsolete
```

Comment:
would remove the TODO. If someone removes the ExecutionState interface he'll have to remove the isKilled() method anyway.